firebaseBackup.js

```diff
@@ -115,9 +124,9 @@ Backup.prototype.backupDB = function(){
           self.compress(fileName).then(function(compressedFileName){
             self.saveS3(compressedFileName, self.params.dbHostName ).then(
               function(){
-                downloadPromise.resolve();
+                downloadPromise.resolve(compressedFileName);
               }, function(){
```

Comment:
I see that you're rejecting it, but you're still treating this as success. The command needs to fail if the S3 upload fails.